src/api/dataops/filteringCondition.spec.ts

```diff
@@ -1,103 +1,248 @@
-    let value = "value";
+    it("creates the proper CompositeFilteringCondition object when using OR", () => {
+      const compCondition = subject.or(createSubject().subject);
 
-    beforeEach( () => {
-      condition = new FilteringCondition( field, operator, [value]);
+      expect(compCondition instanceof CompositeFilteringCondition).toBeTruthy();
     });
 
-    it("creates the proper CompositeFilteringCondition object", () => {
-      let compCondition = condition.or(new FilteringCondition(field, operator, ["value2"]));
-      expect(compCondition.compile()).toEqual({
-        conditions: [
-          {field, operator, values: [value], type: "and"},
-          {field, operator, values: ["value2"], type: "or"}],
-        type: "and"});
+    it("creates the proper CompositeFilteringCondition object when using AND", () => {
+      const compCondition = subject.and(createSubject().subject);
+
+      expect(compCondition instanceof CompositeFilteringCondition).toBeTruthy();
+    });
+
+    describe('when compiling', () => {
+      it("returns the correct format when using the OR operator", () => {
+        const {
+          field: anotherField,
+          operator: anotherOperator,
+          value: anotherValue,
+          subject: anotherSubject,
+        } = createSubject();
+
+        const compCondition = subject.or(anotherSubject);
+
+        expect(compCondition.compile()).toEqual([
+          { field }, operator, value,
+          "or",
+          { field: anotherField }, anotherOperator, anotherValue,
+        ]);
+      });
+
+      it("returns the correct format when using the AND operator", () => {
+        const {
+          field: anotherField,
+          operator: anotherOperator,
+          value: anotherValue,
+          subject: anotherSubject,
+        } = createSubject();
+
+        const compCondition = subject.and(anotherSubject);
+
+        expect(compCondition.compile()).toEqual([
+          { field }, operator, value,
+          "and",
+          { field: anotherField }, anotherOperator, anotherValue,
+        ]);
+      });
     });
   });
 });
 
+function createCompositeSubject({
+  filtering = createSubject(),
+  type = faker.helpers.randomize(["and", "or"]),
+} = {}) {
+  return {
+    ...filtering,
+    type,
+    compositeSubject: new CompositeFilteringCondition(filtering.subject, (<LogicalOperator> type)),
+  };
+}
+
 describe("CompositeFilteringCondition class", () => {
 
   it("should use the given logical operator type", () => {
-    const compCondition = new CompositeFilteringCondition(
-      new FilteringCondition("field", "operator", ["value"]), "and");
-    expect(compCondition.Type).toBe("and");
+    const { compositeSubject, type } = createCompositeSubject();
+    expect(compositeSubject.type).toBe(type);
+  });
+
+  describe("when calling or()", () => {
+    it("should not mutate condition", () => {
+      const { compositeSubject, field, operator, value } = createCompositeSubject();
```

Comment:
You do not use field, operator and value, do you?

Reply:
no. fixed